Update documentation for google-java-format minimum version; not 1.8 but 1.10

- summary of problem: Documentation is outdated for google-java-format version
- Maven version 3.6.3
- spotless version 2.43.0
- operating system Windows 10
- Spotless configuration
```xml
                <configuration>
                    <formats>
                        <!-- you can define as many formats as you want, each is independent -->
                        <format>
                            <includes>
                                <include>src/main/java/**/*.java</include>
                                <include>src/test/java/**/*.java</include>
                            </includes>
                            <!-- define the steps to apply to those files -->
                        </format>
                    </formats>
                    <!-- define a language-specific format -->
                    <java>
                        <googleJavaFormat>
                            <version>1.8</version>                      <!-- optional, 1.8 is minimum supported version -->
                            <style>GOOGLE</style>                       <!-- or AOSP (optional) -->
                            <reflowLongStrings>true</reflowLongStrings> <!-- optional -->
                            <formatJavadoc>false</formatJavadoc>        <!-- optional -->
                            <!-- optional: custom group artifact (you probably don't need this) -->
                            <groupArtifact>com.google.googlejavaformat:google-java-format</groupArtifact>
                        </googleJavaFormat>
                    </java>
                </configuration>
```
- copy-paste the full content of any console errors emitted by `gradlew spotless[Apply/Check] --stacktrace`
  - -- stacktrace is no option. Its *-X,--debug  &rarr; Produce execution debug output*
  - I think the output is clear and just the documentation has to be updated
```log
  [ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.43.0:check (default-cli) on project Spotless: Execution default-cli of goal com.diffplug.spotless:spotless-maven-plugi
n:2.43.0:check failed: You are running Spotless on JVM 17. This requires google-java-format of at least 1.10.0 (you are using 1.8).
```

--- a/plugin-maven/README.md
+++ b/plugin-maven/README.md
@@ -232,7 +232,7 @@ any other maven phase (i.e. compile) then it can be configured as below;
 
 ```xml
 <googleJavaFormat>
-  <version>1.8</version>                      <!-- optional, 1.8 is minimum supported version -->
+  <version>1.8</version>                      <!-- optional, 1.8 is the minimum supported version for Java 11-->
   <style>GOOGLE</style>                       <!-- or AOSP (optional) -->
   <reflowLongStrings>true</reflowLongStrings> <!-- optional -->
   <formatJavadoc>false</formatJavadoc>        <!-- optional -->

